assets/js/components/ExchangeRates.js

```diff
@@ -1,106 +1,175 @@
+    const [calculateValue, setCalculateValue] = useState('');
+    const [calculateBuy, setCalculateBuy] = useState(-1);
+    const [calculateSell, setCalculateSell] = useState(-1);
+    const [calculateExchange, setCalculateExchange] = useState('');
+    const [calculateCode, setCalculateCode] = useState('');
 
-    getBaseUrl() {
-        return 'http://telemedi-zadanie.localhost';
-    }
+    const getBaseUrl = () => 'http://telemedi-zadanie.localhost';
 
-    componentDidMount() {
+    useEffect(() => {
         const urlParams = new URLSearchParams(window.location.search);
-        const date = urlParams.get('date')
-        if (date !== null) {
-            this.state.date = date
+        const dateParam = urlParams.get('date');
+        if (dateParam !== null) {
+            setDate(dateParam);
         }
+        fetchExchangeRates();
+    }, []);
 
-        this.fetchExchangeRates();
-    }
+    useEffect(() => {
+        recalculate();
+        recalculateCode();
+    }, [calculateValue, calculateBuy, calculateSell, rates]);
 
-    fetchExchangeRates() {
-        const baseUrl = this.getBaseUrl();
-        const date = this.state.date;
-        axios.get( `${baseUrl}/api/exchange-rates?date=${date}`).then(response => {
-            this.setState({
-                rates: response.data,
-                loading: false
-            })
-        }).catch(function (error) {
+    const fetchExchangeRates = () => {
+        const baseUrl = getBaseUrl();
+        axios.get(`${baseUrl}/api/exchange-rates?date=${date}`).then(response => {
+            setRates(response.data);
+            setLoading(false);
+        }).catch(error => {
             console.error(error);
-            this.setState({ rates: false, loading: false});
+            setRates(false);
+            setLoading(false);
         });
-    }
+    };
 
-    changeDate = (event) => {
+    const changeDate = (event) => {
         const formattedDate = event.target.value;
-        this.setState({date: formattedDate})
+        setDate(formattedDate);
         window.history.replaceState(null, null, `?date=${formattedDate}`);
-        this.fetchExchangeRates()
-    }
+        fetchExchangeRates();
+    };
+
+    const setExchangeValue = (event) => {
+        setCalculateValue(event.target.value);
+    };
+
+    const setExchangeSell = (event) => {
+        setCalculateSell(event.target.value);
+    };
+
+    const setExchangeBuy = (event) => {
+        setCalculateBuy(event.target.value);
+    };
+
+    const recalculate = () => {
+        if (Number(calculateValue) < 0 || calculateSell === -1 || calculateBuy === -1) {
+            return;
+        }
+
+        if (rates[calculateBuy].buy === 0) {
+            setCalculateExchange('Kantor nie prowadzi sprzedaży tej waluty')
+            setCalculateCode('')
+            return;
+        }
+
+        const sellCalc = rates[calculateSell].sell * Number(calculateValue);
+        const buyCalc = sellCalc / rates[calculateBuy].buy;
```

Comment:
Nie widze byś do klakulacji stosował modifikatory kantoru (czyli te -0.05 albo 0.17), to jest gdzieś indzej?